### DISCUSSION TITLE
set a system prompt/message for queryengine

### DISCUSSION THREAD

**Discussion by rawwerks** [run-llama/LlamaIndexTS#1617] [Q&A]
i cannot for the life of me figure out how to set a system prompt while using index.asQueryEngine()

could someone point me in the right direction? this has to be possible, otherwise how could anyone use llamaindexts for RAG applications?

here's a related discord thread: https://discord.com/channels/1059199217496772688/1133167189860565033/1214393825473077248

**Comment by marcusschiesser** [comment]
a query engine is using a response synthesizer (see https://ts.llamaindex.ai/docs/llamaindex/modules/query_engines) to generate the response.
So you'll need to modify the prompt of the response synthesizer, see https://ts.llamaindex.ai/docs/llamaindex/modules/prompt
